Upgrading  NooBaa from centos7 to centos8

liranmauda opened the pull request:
## Upgrading  NooBaa from centos7 to centos8

- Added .eslintignore for skipping running eslint on the ceph tests directory
- Added set_mongo_repo.sh for adding a mongo 3.6 yum repo

builder.Dockerfile:
- Change  centos:7 to centos:8
- Removed the centos-release-scl
- Removed the devtoolset-7 and it's calls
- Replaced the yum with dnf
- Setting alternatives python

Base.Dockerfile:
- Remove the use of devtoolset-7

NooBaa.Dockerfile:
- Change  centos:7 to centos:8
- Removed the use of devtoolset-7
- Update rsyslog from 8.24.0 to 8.37.0
- Update python-setuptools to python2-setuptools

Tests.Dockerfile:
- Removed centos-release-scl
- Copying the set_mongo_repo.sh and .eslintignore into the container
- Installing the mongo from the added yum repo
- Updated the ceph test pre-requirements.
- Using dnf instead of yum

Attributes.Dockerfile:
- Removed the calls for devtoolset-7

setup_platform.sh:
- Calling easy_install-2 instead of easy_install

.eslintrc.js:
- Fix typos

Makefile:
- Painting the "done" message in green now.

run_npm_test_on_test_container.sh:
- Removed unneccessery call for enabling the mongo

liranmauda commented:
Currently, travis will fail due to https://github.com/ceph/s3-tests/issues/318

mykaul commented on src/deploy/NVA_build/Tests.Dockerfile:
```diff
@@ -13,11 +13,20 @@ ENV TEST_CONTAINER true
+RUN chmod +x /tmp/set_mongo_repo.sh && \
+    /bin/bash -xc "/tmp/set_mongo_repo.sh"
+
+RUN dnf install -y -q vim \
```
Why do we need vim?!

guymguym replied:
To edit scripts inside a running pod. Allows us to make patches inline very quickly to test assumptions or fix complex situations.

guymguym replied:
I can't even start to explain how valuable this is.

liranmauda replied:
The vim is only on the test container, it will not affect the core image. 
on the test container, we extend the noobaa-core image and adding some tools for easier debug + tools for mocking our flows (such as adding mongo to the image) so we will be able to run all the unit tests as a single docker run (without relying on and kubernetes env)

liranmauda committed:
Upgrading NooBaa from centos7 to centos8

- Added .eslintignore for skipping running eslint on the ceph tests directory
- Added set_mongo_repo.sh for adding a mongo 3.6 yum repo

builder.Dockerfile:
- Change  centos:7 to centos:8
- Removed the centos-release-scl
- Removed the devtoolset-7 and it's calls
- Replaced the yum with dnf
- Setting alternatives python

Base.Dockerfile:
- Remove the use of devtoolset-7

NooBaa.Dockerfile:
- Change  centos:7 to centos:8
- Removed the use of devtoolset-7
- Update rsyslog from 8.24.0 to 8.37.0
- Update python-setuptools to python2-setuptools

Tests.Dockerfile:
- Removed centos-release-scl
- Copying the set_mongo_repo.sh and .eslintignore into the container
- Installing the mongo from the added yum repo
- Updated the ceph test pre-requirements.
- Using dnf instead of yum

Attributes.Dockerfile:
- Removed the calls for devtoolset-7

setup_platform.sh:
- Calling easy_install-2 instead of easy_install

run_npm_test_on_test_container.sh:
- Removed unneccessery call for enabling the mongo

liranmauda committed:
Small changes in makefile and typo fixes.

.eslintrc.js:
- Fix typos

Makefile:
- Painting the "done" message in green now.

liranmauda committed:
Update ceph_s3_tests_deploy.sh
